test/chainrules.jl

```diff
@@ -14,4 +15,13 @@
         v = rand(6)
         test_rrule(*, D, v)
     end
+
+    @testset "ProjectTo" begin
+        bd = BlockDiagonal([ones(2, 2), ones(3, 3)])
+        project = ProjectTo(bd)
+        @test project(ones(5, 5)) == bd
+        @test project(reshape(ones(5, 5), 5, 5)) == bd
```

Comment:
What about `project(bd)`? 

Also, is this not identical to the previous test i.e. `reshape(ones(5, 5), 5, 5) == ones(5,5)`?

Reply:
ah nice catch, I wanted to get some kind of wrapper array with a `parent` (now using adjoint instead)